extensions/amp-story/0.1/system-layer.js

```diff
@@ -0,0 +1,258 @@
+  /**
+   * @param {!Window} win
+   */
+  constructor(win) {
+    /** @private {!Window} */
+    this.win_ = win;
+
+    /** @private {boolean} */
+    this.isBuilt_ = false;
+
+    /** @private {?Element} */
+    this.root_ = null;
+
+    /** @private {boolean} */
+    this.inFullScreen_ = false;
+
+    /** @private {?Element} */
+    this.exitFullScreenBtn_ = null;
+
+    /** @private {?Element} */
+    this.closeBookendBtn_ = null;
+
+    /** @private {?Element} */
+    this.muteAudioBtn_ = null;
+
+    /** @private {?Element} */
+    this.unmuteAudioBtn_ = null;
+
+    /** @private @const {!ProgressBar} */
+    this.progressBar_ = ProgressBar.create(win);
+  }
+
+  /**
+   * @param {number} pageCount The number of pages in the story.
+   * @return {!Element}
+   */
+  build(pageCount) {
+    if (this.isBuilt_) {
+      return this.getRoot();
+    }
+
+    this.isBuilt_ = true;
+
+    this.root_ = this.win_.document.createElement('aside');
+    this.root_.classList.add('i-amphtml-story-system-layer');
+    this.root_./*OK*/innerHTML = TEMPLATE;
+
+    this.root_.insertBefore(
+        this.progressBar_.build(pageCount), this.root_.firstChild);
+
+    this.exitFullScreenBtn_ =
+        this.root_.querySelector('.i-amphtml-story-exit-fullscreen');
+
+    this.closeBookendBtn_ =
+        this.root_.querySelector('.i-amphtml-story-bookend-close');
+
+    this.muteAudioBtn_ =
+        this.root_.querySelector('.i-amphtml-story-mute-audio-control');
+
+    this.unmuteAudioBtn_ =
+        this.root_.querySelector('.i-amphtml-story-unmute-audio-control');
+
+    this.progressEl_ =
+        this.root_.querySelector('.i-amphtml-story-progress-value');
+
+    this.addEventHandlers_();
+
+    return this.getRoot();
+  }
+
+  /**
+   * @private
+   */
+  addEventHandlers_() {
+    // TODO(alanorozco): Listen to tap event properly (i.e. fastclick)
+    this.exitFullScreenBtn_.addEventListener(
+        'click', e => this.onExitFullScreenClick_(e));
+
+    this.closeBookendBtn_.addEventListener(
+        'click', e => this.onCloseBookendClick_(e));
+
+    this.muteAudioBtn_.addEventListener(
+        'click', e => this.onMuteAudioClick_(e));
+
+    this.unmuteAudioBtn_.addEventListener(
+        'click', e => this.onUnmuteAudioClick_(e));
+  }
+
+  /**
+   * @return {!../../../src/service/vsync-impl.Vsync}
+   * @private
+   */
+  getVsync_() {
+    return Services.vsyncFor(this.win_);
+  }
+
+  /**
+   * @return {!Element}
+   */
+  getRoot() {
+    return dev().assertElement(this.root_);
+  }
+
+  /**
+   * @param {boolean} inFullScreen
+   */
+  setInFullScreen(inFullScreen) {
+    this.inFullScreen_ = inFullScreen;
+
+    this.toggleExitFullScreenBtn_(inFullScreen);
+  }
+
+  /**
+   * @param {boolean} isEnabled
+   * @private
+   */
+  toggleExitFullScreenBtn_(isEnabled) {
+    this.getVsync_().mutate(() => {
+      toggleHiddenAttribute(
+          dev().assertElement(this.exitFullScreenBtn_),
+          /* opt_isHidden */ !isEnabled);
+    });
```

Comment:
Minor: Putting `mutate()` inside `toggleHiddenAttribute()` will make sure it's not called outside of a mutate context.

Reply:
Done